### PR TITLE
fix(vue-app): prevent double layout execution (#5703)

### PR DESCRIPTION
**Pull request by galvez**
In vue-app/client.js, we use `showNextPage()` to set layout, which is ran in an `afterEach` hook. As reported in #5703, this causes a double execution when switching layouts. Perhaps having this occur in `afterEach` is not fast enough -- switching to `beforeResolve` seems to solve it.

Update: using `afterEach` works, but the actual call must happen before `nextTick`.

### CODE REVIEW

File: packages/vue-app/template/client.js [hunk 1]
```diff
@@ -611,7 +606,15 @@ function showNextPage (to) {
     layout = layout(app.context)
   }
   this.setLayout(layout)
-  <% } %>
+  next()
+}
+<% } %>
+
+function checkForErrors (to) {
```
Comment (pi0) [inline packages/vue-app/template/client.js:613]: to argument is unused. Can we use (app) without binding this?

Comment (galvez) [reply packages/vue-app/template/client.js:613]: Sure.

Comment (galvez) [reply packages/vue-app/template/client.js]: Let me know if it's okay now.

Comment (galvez) [reply packages/vue-app/template/client.js]: Looks like also we got a CI error but seems to be unrelated?

**Comment by galvez** [comment]
Also reported in #7441.

**Comment by pimlie** [comment]
Does this change influence page transitions? Did you check manually? Dont think we have proper tests to make sure they perform well visually?

But Im worried that this might be a breaking change (albeit it probably a low risk one). If a user already adds their own beforeResolve hook _and_ cancels navigation (or redirects) within that, then the new/wrong layout will already be set.

**Comment by galvez** [comment]
I did test manually, seems ok.

Re: `beforeResolve()`, I had an idea to workaround this, will try it next.

**Comment by galvez** [comment]
@pimlie turns out we can use `afterEach` for this, we just can't `await` for `nextTick`.

**Comment by simplenotezy** [comment]
🎉🎉

**Review by atinux — APPROVED** [review]

**Comment by simplenotezy** [comment]
Great!! Good job.

@Atinux Can I pull this fix in through nuxt-edge or how can I get this fix in my application?

**Comment by pi0** [comment]
@simplenotezy `nuxt-edge` releases are nightly so you should be able to test if it is fixing issue in your application.

**Comment by simplenotezy** [comment]
I see, thanks @pi0

**Comment by simplenotezy** [comment]
I can confirm it's working on my end. Good job guys

**Comment by ademyalcin27** [comment]
Hi,
I was using Nuxt@2.9.2 and which is mounted triggered 2 times in mobile. after that I upgraded my nuxt version from 2.9.2 to 2.13.3 I still got same error, Also It's same layout calling 2 times in mounted, '**VirtualPageViewAll**', and '**mobileDefault**' calling in mounted

Ekran Resmi 2020-07-07 13 42 15

**Comment by hjujah** [comment]
For me its not page but all page components are mounting twice. Tried Nuxt 2.13 and 2.14 and edge (2.15) but having the same issue everywhere. Please is there any fix for this?

**Pull request merged**